### DISCUSSION TITLE
Multiple export action in a page doesn't work

### DISCUSSION THREAD

**Discussion by DavidKoLab** [filamentphp/filament#12913] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2.80 and v3.2.82

### PHP Version
8.3

### Laravel Version
11

### How can we help you?

I tried to place 2 export action buttons especting to get 2 kind of reports. one for vendors and another one for users, eventhough 2 buttons appeared in the view, the exportUserAction rendered btn still run the code in VendorExporter instead of code in exportUserAction. I tried to add a parameter to make() method but unfortunately when ExportAction::make() receive a string it made the whole button behave like disabled button. 

```
class Reports extends Page{
...
 public function exportAction()
    {
        return ExportAction::make()
        ->label('匯出廠商資料')
        ->exporter(VendorExporter::class)
        ->modifyQueryUsing(fn (Builder $query, array $data) => $query->whereBetween('created_at', [$data['start_date'], $data['end_date']])//data is the result of inputs defined in VendorExporter::getOptionsFormComponents()
        );
    }


    public function exportUserAction(){
        return ExportAction::make()
        ->label('匯出使用者資料')
        ->exporter(UserExporter::class)
        ->modifyQueryUsing(fn (Builder $query, array $data) => $query->whereBetween('created_at', [$data['start_date'], $data['end_date']])//data is the result of inputs defined in UserExporter::getOptionsFormComponents()
        );
    }
   .....
}
```

view
```
<x-filament-panels::page>
        <x-filament::section>
            <x-slot name="heading">
                請選擇要生成的報表
            </x-slot>
            {{ $this->exportAction }}
            {{-- Content --}}
            {{$this->exportUserAction}}
            {{$this->testAction}}
        </x-filament::section>
</x-filament-panels::page>

```

**Comment by dissto** [comment]
You probably need to give each action a unique name? 

```php
return ExportAction::make('vendor_export')
->....

return ExportAction::make('user_export')
->...

```
?

**Reply by DavidKoLab** [reply]
When names are set in make method, the btns lose their functionalities. It behaves different from common actions. It seems the only way to make one button work is to set ExportAction::make() or ExportAction::make('export'), any other name will make the button not functioning. (still sending the .../livewire/update request to the server but it won't open the modal as it should.) So it is impossible to have unique names since the only name that works is export (when no passing anything to make, it will use name 'export' by default.)

**Reply by dissto** [reply, marked as answer]
Well I think you should also name the functions accordingly. 

```php
Public function userExportAction()
{
   ExportAction::make('userExport')
...
}
```
If that makes sense.

https://filamentphp.com/docs/3.x/actions/adding-an-action-to-a-livewire-component#adding-the-action

**Reply by DavidKoLab** [reply]
@dissto You're right, I didn't check that part.
